### PR TITLE
fix(pcc.user): fix pagination all user not displayed

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pcc-federation` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-10106 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I  fixed the pagination all users aren't displayed

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/dedicatedCloud.service.js [hunk 1]
```diff
@@ -560,7 +560,7 @@ class DedicatedCloudService {
       .query()
       .expand('CachedObjectList-Pages')
       .limit(pageSize)
-      .offset(offset)
+      .offset(Math.ceil(offset / pageSize))
```
Comment (frenautvh) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/dedicatedCloud.service.js:563]: pageSize || 1

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/vsphere-users/vsphere-users.controller.js [hunk 2]
```diff
@@ -58,21 +58,19 @@ export default class DedicatedCloudVsphereUsersCtrl {
       defaultFilterColumn,
     };
 
-    return this.$q
-      .resolve(this.DedicatedCloud.getUsers(this.productId, params))
-      .then((users) => {
-        const userFormat = users.map((user) => {
+    return this.DedicatedCloud.getUsers(this.productId, params).then(
+      ({ data, meta }) => {
+        const userFormat = data.map((user) => {
```
Comment (mohammed-zahaf) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/vsphere-users/vsphere-users.controller.js:63]:
```suggestion
      ({ data: users, meta }) => {
        const userFormat = users.map((user) => {
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/dedicatedCloud.service.js (hunk 1 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/dedicatedCloud.service.js]: It's applied.

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/vsphere-users/vsphere-users.controller.js (hunk 2 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/vsphere-users/vsphere-users.controller.js]: It's applied

**Review by mohammed-zahaf — APPROVED** [review]

**Pull request merged**